types/webpack-dev-server/index.d.ts

```diff
@@ -82,6 +80,12 @@ declare class WebpackDevServer {
         config: WebpackDevServer.Configuration
     );
 
+    static addDevServerEntrypoints(
+        webpack: webpack.Compiler | webpack.MultiCompiler,
```

Comment:
@alan-agius4 Thank you. One extra finding... according to https://github.com/webpack/webpack-dev-server/blob/master/lib/util/addDevServerEntrypoints.js#L22, the `webpackOptions` has to be an array... `webpack.Compiler | webpack. MultiCompiler` shouldn't be the right type...

Ref from  `webpack`'s definition:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/webpack/index.d.ts#L533
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/webpack/index.d.ts#L560

Ref from `webpack-dev-server`'s implementation:
https://github.com/webpack/webpack-dev-server/blob/master/lib/util/addDevServerEntrypoints.js#L22-L32

Reply:
you are right that it `webpackOptions`  but it doesnt need to be an array.
https://webpack.js.org/guides/hot-module-replacement/#via-the-node-js-api

And also that is why there is `[].concat(webpackOptions)` at https://github.com/webpack/webpack-dev-server/blob/master/lib/util/addDevServerEntrypoints.js#L22  so that if it's not an array it will be casted to an array

Reply:
good catch.